### PR TITLE
Updated edx deploy to support custom instance sizes

### PR DESCRIPTION
**Pull request by blarghmatey**

### CODE REVIEW

**Comment by shaidar** [comment]
Just making sure that the elasticsearch changes are intentionally part of this

File: salt/orchestrate/aws/map_templates/edx.yml [hunk 1]
```diff
@@ -10,10 +10,11 @@
 {% set num_subnets = subnetids|length %}
 
 edx:
-  {% for app_type, num_instances in app_types.items() %}
-  {% for id_num in range(num_instances.edx) %}
+  {% for app_type, settings in app_types.items() %}
```
Comment (shaidar) [inline salt/orchestrate/aws/map_templates/edx.yml]: Where is `app_types` defined?

Comment (blarghmatey) [reply salt/orchestrate/aws/map_templates/edx.yml]: It gets passed in as a context object from the orchestrate state

**Comment by blarghmatey** [comment]
The elastic search changes were not intended to be included, but they have been removed as part of rebasing against master.

File: salt/orchestrate/aws/map_templates/edx.yml (hunk 1 above)
Comment (shaidar) [reply salt/orchestrate/aws/map_templates/edx.yml]: There is no longer a need for [VPC_RESOURCE_SUFFIX](https://github.com/mitodl/salt-ops/blob/d5a0dcd7359b7e826c29a24a9c40f416cf808e0c/salt/orchestrate/aws/map_templates/edx.yml#L1) correct?

File: salt/orchestrate/edx/deploy.sls [hunk 2]
```diff
@@ -66,9 +73,14 @@ generate_edx_cloud_map_file:
...
         profile_overrides:
           userdata_file: '/etc/salt/cloud.d/edx_userdata.yml'
+          size: {{ purposes['{}-live'.format(PURPOSE_PREFIX)].instance_type }}
         app_types:
-          draft: {{ purposes['{}-draft'.format(PURPOSE_PREFIX)].num_instances }}
-          live:  {{ purposes['{}-live'.format(PURPOSE_PREFIX)].num_instances }}
+          draft:
+            num_instances: {{ purposes['{}-draft'.format(PURPOSE_PREFIX)].num_instances }}
+            instance_type: {{ purposes['{}-draft'.format(PURPOSE_PREFIX)].instance_type }}
+            live:
```
Comment (shaidar) [inline salt/orchestrate/edx/deploy.sls]: Is this the right spacing or should this line up with `draft`?

File: salt/orchestrate/aws/map_templates/edx.yml (hunk 1 above)
Comment (blarghmatey) [reply salt/orchestrate/aws/map_templates/edx.yml]: Correct

**Comment by shaidar** [comment]
👍

**Pull request merged**